### ISSUE TITLE
terminal-notifier style notification never shows

### ISSUE THREAD

**Issue by asimpson** [jwiegley/alert#42]
terminal-notifier style notification is never displayed _but_ is added to the Notification Center in OS X. The behavior is like I have Do Not Disturb enabled even when I don't. A notification does correctly display when I invoke `terminal-notifier` directly via `async-shell-command`.

**Comment by asimpson** [comment]
Update: I've tracked this down to the `-sender` flag. Removing this causes the notification to fire correctly.

**Comment by waisbrot** [comment]
Well! I was having this problem also, but it turns out not to be a problem. Sort of.

The issue is that when you identify the alert sender as Emacs, Mac OS will not display the alert when Emacs is in the foreground. This is consistent with all other Mac alerts. If you do something like

```
(progn (sleep-for 3)
       (alert "hello world"))
```

And then tab away from Emacs, I believe you'll see the alert working correctly.

**Comment by asimpson** [comment]
@waisbrot ah that makes sense, thank you! I'm going to have to figure out another way to be alerted for `org` related things when I'm _in_ Emacs.

**Comment by waisbrot** [comment]
I was thinking about this last night. The problem is that while the Mac rule makes sense for normal applications (don't alert me about a new email when I'm in the middle of reading email), Emacs is more like an operating system here. I **do** want to be alerted about a new email when I'm editing some source code.

Therefore, it seems to me that what you and I want *is* to take away `-sender` and have Emacs manage the logic of "display an alert only if it's coming from an inactive buffer".

**Comment by asimpson** [comment]
Yep, that’s where I’ve landed as well. It might also make sense to explore a more “native” to Emacs notification approach, but I don’t know what that looks like at the moment.

**Comment by seanfarley** [comment]
So, alert hasn't been popping up for me either. I've been using some amalgamation of sauron / alert and recently got rid of sauron so I thought it was that. After some googling and banging my head against the wall, I find this issue. Great! Other people have this problem.

So, I go to look at which asshole added the `-sender` flag: 556c6300

**Comment by asimpson** [comment]
:joy: Yeah that guy, what a moron :joy:

**Comment by jwiegley** [comment]
I'll revert it.

**Comment by seanfarley** [comment]
Well, before it's reverted, we need to decide how clicking on a notification will bring up emacs, right?

**Comment by jwiegley** [comment]
It's better to revert, get back functionality for *reporting* notifications, and then decide that question.  I never click on notifications from alert.el, so I hadn't even thought about the usefulness of this: although, it would be great to click on the bubble, and jump to the buffer that originating the warning.

**Comment by seanfarley** [comment]
Righto, totally agree. Though, just to point out, it's only for notifications when emacs is active (still works with `-sender` when not active ... didn't know if that was clear or not).
